src/zenml/materializers/pandas_series_materializer.py

```diff
@@ -0,0 +1,78 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import os
+import tempfile
+from typing import Any, Type
+
+import pandas as pd
+
+from zenml.artifacts import DataArtifact, SchemaArtifact, StatisticsArtifact
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+DEFAULT_FILENAME = "df.parquet.gzip"
+COMPRESSION_TYPE = "gzip"
+
+
+class PandasSeriesMaterializer(BaseMaterializer):
+    """Materializer to read data to and from pandas Series."""
+
+    ASSOCIATED_TYPES = (pd.Series,)
+    ASSOCIATED_ARTIFACT_TYPES = (
+        DataArtifact,
+        StatisticsArtifact,
+        SchemaArtifact,
+    )
+
+    def handle_input(self, data_type: Type[Any]) -> pd.Series:
```

Comment:
Rather than doing this all again, how about just calling the PandasMaterializer here

```
from zenml.materializers.pandas_materializer import PandasMaterializer

df = PandasMaterializer(self.artifact).handle_input(data_type)
```

Reply:
@htahir1 sent me a template for this materializer that took that approach. When testing the code I ran into errors I couldn't resolve. Here is a picture of the most common error I got.
![Screenshot from 2022-06-02 00-46-39](https://user-images.githubusercontent.com/3477188/171832754-ae4a320d-07b7-48e8-8925-c57332fe9e74.png)